src/pages/Navgurukul/merakiAdmission/index.js

```diff
@@ -0,0 +1,211 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { toast } from "react-toastify";
+import YouTube from "react-youtube";
+import { METHODS } from "../../../services/api";
+import "./styles.scss";
+
+function Admission() {
+  const [userDetails, setUserDetails] = useState({
+    firstName: "",
+    middleName: "",
+    lastName: "",
+    mobileNumber: "",
+  });
+
+  const [mobile, setMobile] = useState("");
+  const [enrolmentKey, setEnrolmentKey] = useState("");
+
+  const [partnerId, setPartnerId] = useState("");
+
+  let userToken = localStorage.getItem("Token");
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/users/me`,
+      headers: {
+        accept: "application/json",
+        Authorization: userToken,
+      },
+    }).then((res) => {
+      setPartnerId(res.data.user.partner_id);
+    });
+  }, []);
+
+  const testUrl = "http://dev-join.navgurukul.org/k/";
+
+  const generateTestLink = async () => {
+    try {
+      const mobile = `0${userDetails.mobileNumber}`;
+      const dataURL = `${process.env.REACT_APP_CHANAKYA_BASE_URL}helpline/register_exotel_call`;
+      const response = await axios.get(dataURL, {
+        params: {
+          ngCallType: "getEnrolmentKey",
+          From: mobile,
+          partner_id: partnerId ? partnerId : null,
+        },
+      });
+      setEnrolmentKey(response.data.key);
+      const params = {
+        firstName: userDetails.firstName,
+        middleName: userDetails.middleName,
+        lastName: userDetails.lastName,
+        mobileNumber: userDetails.mobileNumber,
+      };
+      const queryString = Object.keys(params)
+        .map((filter) => {
+          if (params[filter]) {
+            return `${filter}=${params[filter]}`;
+          }
+          return null;
+        })
+        .filter((item) => item)
```

Comment:
Reversing the `map` and `filter` here removes the need for the `if` (and `return null`).
```suggestion
      const queryString = Object.keys(params)
          .filter((key) => params[key])
          .map((key) => `${key}=${params[key]}`);
```